modules/external-api/src/main/java/org/opencastproject/external/endpoint/ListProviderEndpoint.java

```diff
@@ -94,26 +93,28 @@ void activate(ComponentContext cc) {
 
   @GET
   @Path("providers.json")
-  @Produces(MediaType.APPLICATION_JSON)
+  @Produces({ ApiMediaType.JSON, ApiMediaType.VERSION_1_10_0 })
   @RestQuery(name = "availableProviders", description = "Provides the list of the available list providers", responses = { @RestResponse(description = "Returns the availables list providers.", responseCode = HttpServletResponse.SC_OK) }, returnDescription = "")
-  public Response getAvailablesProviders(@Context HttpHeaders headers) {
+  public Response getAvailablesProviders(@HeaderParam("Accept") String acceptHeader) {
+    final ApiVersion requestedVersion = ApiMediaType.parse(acceptHeader).getVersion();
     JSONArray list = new JSONArray();
 
     list.add(listProvidersService.getAvailableProviders());
 
-    return Response.ok(list.toString()).build();
+    return ApiResponses.Json.ok(requestedVersion, list.toJSONString());
```

Comment:
... you can just pass the header directly:

```suggestion
    return ApiResponses.Json.ok(acceptHeader, list.toJSONString());
```